bigbluebutton-html5/imports/ui/components/whiteboard/shapes/poll/component.jsx

```diff
@@ -1,14 +1,569 @@
 import React, { PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import ShapeHelpers from '../helpers.js';
 
 export default class PollDrawComponent extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      //flag indicating whether we need to continue calculating the sizes or display the shape
+      prepareToDisplay: true,
+
+      //outer (white) rectangle's coordinates and sizes (calculated in componentWillMount)
+      outerRect: {
+        x: 0,
+        y: 0,
+        width: 0,
+        height: 0,
+      },
+
+      //inner rectangle's coordinates and sizes
+      innerRect: {
+        x: 0,
+        y: 0,
+        width: 0,
+        height: 0,
+      },
+      thickness: 0,
+      backgroundColor: '#ffffff',
+
+      //max line sizes
+      maxLineWidth: 0,
+      maxLineHeight: 0,
+
+      //max widths of the keys (left) and percent (right) strings
+      maxLeftWidth: 0,
+      maxRightWidth: 0,
+
+      //these parameters are used in calculations before and while displaying the final result
+      votesTotal: 0,
+      maxNumVotes: 0,
+      textArray: [],
+      maxDigitWidth: 0,
+      maxDigitHeight: 0,
+
+      //start value used for font-size calculations
+      calcFontSize: 50,
+      currentLine: 0,
+      lineToMeasure: [],
+      fontSizeDirection: 1,
+    };
+  }
+
+  componentWillMount() {
+    //in this part we retrieve the props and perform initial calculations for the state
+    //calculating only the parts which have to be done just once and don't require
+    //rendering / rerendering the text objects
+
+    //x1 and y1 - coordinates of the top left corner of the shape
+    //initial width and height are the width and height of the shape
+    //all the points are given as percentages of the slide
+    let x1 = this.props.shape.points[0];
+    let y1 = this.props.shape.points[1];
+    let initialWidth = this.props.shape.points[2];
+    let initialHeight = this.props.shape.points[3];
+    let x;
+    let y;
+    let width;
+    let height;
+    let innerWidth;
+    let innerHeight;
+    let innerX;
+    let innerY;
+    let votesTotal;
+    let maxNumVotes;
+    let textArray;
+    let maxLineWidth;
+    let maxLineHeight;
+    let thickness;
+    let arrayLength;
+    let i;
+    let lineToMeasure;
```

Comment:
not to nit pick but your link doesn't actually say you need to post them all at the highest scope possible. Just says break them up.

In fact the link even says declare them when used. So you declare them on the same line as defined (if possible by scope). It doesnt say declare all at the top, and then define on another line (unless you have to). The x and y for example are declared with no value, and defined in the same scope. Which is against what is in the link you posted